packages/components/src/slot-fill/types.ts

```diff
@@ -0,0 +1,119 @@
+/**
+ * External dependencies
+ */
+import type { Component, MutableRefObject, ReactNode } from 'react';
+
+export type SlotKey = string | symbol;
+
+type SlotPropBase = {
+	/**
+	 * Slot name.
+	 */
+	name: SlotKey;
+	/**
+	 * props to pass from `Slot` to `Fill`.
+	 *
+	 * @default {}
+	 */
+	fillProps?: any;
+
+	/**
+	 * By default, events will bubble to their parents on the DOM hierarchy (native event bubbling).
+	 * If set to true, events will bubble to their virtual parent in the React elements hierarchy instead,
+	 * also accept an optional `className`, `id`, etc.  to add to the slot container.
+	 */
+	bubblesVirtually?: boolean;
+};
+
+export type SlotComponentProps =
+	| ( SlotPropBase & {
+			bubblesVirtually: true;
+	  } )
+	| ( SlotPropBase & {
+			bubblesVirtually?: false;
+
+			/**
+			 * A function that returns nodes to be rendered.
+			 * Not supported when `bubblesVirtually` is true.
+			 *
+			 * @param fills
+			 */
+			children?: ( fills: ReactNode ) => ReactNode;
+	  } );
+
+export type FillComponentProps = {
+	/**
+	 * Slot name.
```

Comment:
Maybe a bit more helpful?

```suggestion
	 * The name of the slot to fill into.
```